src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -265,6 +269,8 @@ private void ValidateBundleRequirements(ExtensionBundleDetails bundleDetails)
                 if (bundleVersion < minimumVersion)
                 {
                     _logger.MinimumBundleVersionNotSatisfied(bundleDetails.Id, bundleDetails.Version, requirement.MinimumVersion);
+                    // Add tag? Or use health check instead of metric?
+                    _hostMetric.AppFailure();
```

Comment:
I'm thinking a failure here will already be caught and counted by the main startup loop you've already instrumented. Same for the other instances in this file.